dsl.go

```diff
@@ -330,10 +339,15 @@ func (stubbing *Stubbing) Invoke(params []Param) ReturnValues {
 
 type Matchers []Matcher
 
+var matchersMutex sync.Mutex
+
 func (matchers Matchers) Matches(params []Param) bool {
 	if len(matchers) != len(params) { // Technically, this is not an error. Variadic arguments can cause this
 		return false
 	}
+
+	matchersMutex.Lock()
+	defer matchersMutex.Unlock()
```

Comment:
I think we should move this lock into the matchers themselves instead. The matchers have this annoying problem, that they are not thread-safe, because they need to transport the `actual` value from one method to the other, because this is simply how the protocol works. It's a bit silly, but there is no really good way to work around it with the current design (which is from Gomega).

I suggest we remove this locking here, and add it to the AnyMatcher and EqMatcher. What do you think?

Reply:
I think should try and see what happens :-)

Reply:
@petergtz as it turned out, if there are no invocations and verifications concurrently, then there is no need to fix Matcher

Reply:
Ops, data race when tested in travis-ci :-p